docs/resources/kafka_topic.md

```diff
@@ -0,0 +1,151 @@
+---
+page_title: "instaclustr_kafka_topic Resource - terraform-provider-instaclustr"
+subcategory: ""
+description: |-
+  
+---
+
+# Resource  `instaclustr_kafka_topic` and Data Source `instaclustr_kafka_topic_list`
+Resources for managing Kafka topics for a Kafka cluster.
+Kafka topic list is a read-only data source used to get the list of kafka topics in a cluster,
+while Kafka topic is a resource used to create topics, read topics' replication-factor and partitions, 
+read topics' configs, update topics' configs, and delete Kafka topics.
+
+**_Note that we don't allow an update of a topic's replication factor and partitions number._**
+
+## Import
+You can import the existing topics of a Kafka cluster created via other ways (e.g. API, Console) to your
+terraform resources.
+
+### Import a topic and its replication-factor and partitions
+```shell
+terraform import instaclustr_kafka_topic.<resource-name> "<cluster-id>&<topic-name>"
+```
+
+### Import a topic and its configs
+```shell
+terraform import instaclustr_kafka_topic.<resource-name> "<cluster-id>&<topic-name>&config"
```

Comment:
Because there could be cases that our customers only want to manage topics (create/delete) rather than topic configs. Then in this case, I think there is no need to import the topic config.

Reply:
The standard for importing a resource is to have singular behaviour though, not have conditional behaviour based on the input. The input to a terraform config is meant to be an identifier for the resource, not a directive. If the user only cares about the non-config properties, all they need to do is match the config with what is imported, which is a simple thing to do.